test/runtests.jl

```diff
@@ -1,7 +1,149 @@
 using RemoteSemaphores
-using Base.Test
+using RemoteSemaphores: _current_count
+using Compat.Test
+
+using Compat: @schedule
+using Compat.Dates
+
+struct TimeoutException
+    duration
+end
+
+function Base.showerror(io::IO, te::TimeoutException)
+    print(io, "TimeoutException: Operation did not finish in ", te.duration)
+
+    if !isa(te.duration, Period)
+        print(io, " seconds")
+    end
+end
+
+function asynctimedwait(fn, secs; kill=false)
```

Comment:
I have a feeling this works much better than https://github.com/ararslan/Timeout.jl...